src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -249,7 +263,34 @@ private boolean checkApproovee(TransactionViewModel approovee) throws Exception
         if(approovee.getHash().equals(Hash.NULL_HASH)) {
             return true;
         }
-        return approovee.isSolid();
+        return approovee.subtangleStatus() == SubtangleStatus.SOLID;
+    }
+
+    public void propagateInvalidSubtangle(Hash invalidHash, boolean force) throws Exception {
```

Comment:
`invalidSubtangleStatus` is an API function name to check solid milestones.
`propagateInvalidSubtangle` + `subtangleStatus` are confusing names in that context.

maybe worth changing  API function name to reflect it's true meaning.